libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1390,11 +1477,23 @@ internal class MapRouteLine(
                     route.geometry() ?: "",
                     Constants.PRECISION_6
                 )
-
+                val randomId = UUID.randomUUID().toString()
```

Comment:
This still relies on a random generator even if the chances of collision are slim. We do have the means to ensure that a collision will never happen since we do know the number of lines that are present which could be a base for our IDs.

I won't treat it as a blocker, but could you cut a follow-up ticket to address this?

Reply:
According to the UUID docs it is a universally unique identifier. So if it's universally unique there should be no collisions that would produce a duplicate.

Reply:
We can never rule collisions out, even without implementation mistakes. Just as a reference - https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions. Whenever we do have a way to avoid resorting to random numbers, I think we should try doing so. Again, not a blocker.